snapshot/src/test/java/io/zeebe/snapshots/impl/SnapshotChunkReaderTest.java

```diff
@@ -136,78 +125,44 @@ public void shouldSeekToChunk() throws Exception {
-    assertSnapshotChunk(
-        expectedSnapshotChecksum,
-        snapshotChunks.get(1),
-        "file3",
-        "content",
-        persistedSnapshot.getId());
   }
 
   @Test
   public void shouldThrowExceptionOnReachingLimit() {
     // given
-    final var index = 1L;
-    final var term = 0L;
-
-    final var transientSnapshot =
-        persistedSnapshotStore.newTransientSnapshot(index, term, 1, 0).orElseThrow();
-    transientSnapshot.take(
-        p -> takeSnapshot(p, List.of("file3", "file1", "file2"), List.of("content", "this", "is")));
-    final var persistedSnapshot = transientSnapshot.persist().join();
-
-    // when
     final var snapshotChunkReader = persistedSnapshot.newChunkReader();
     while (snapshotChunkReader.hasNext()) {
       snapshotChunkReader.next();
     }
 
-    // then
     assertThat(snapshotChunkReader.nextId()).isNull();
 
+    // when - then
     assertThatThrownBy(snapshotChunkReader::next).isInstanceOf(NoSuchElementException.class);
   }
 
-  private void assertSnapshotChunk(
-      final long expectedSnapshotChecksum,
-      final SnapshotChunk snapshotChunk,
-      final String fileName,
-      final String chunkContent,
-      final String snapshotId) {
-    assertThat(snapshotChunk.getSnapshotId()).isEqualTo(snapshotId);
-    assertThat(snapshotChunk.getChunkName()).isEqualTo(fileName);
-    assertThat(snapshotChunk.getContent()).isEqualTo(chunkContent.getBytes());
-    assertThat(snapshotChunk.getTotalCount()).isEqualTo(4);
-    final var crc32c = new CRC32C();
-    crc32c.update(asByteBuffer(chunkContent));
-    assertThat(snapshotChunk.getChecksum()).isEqualTo(crc32c.getValue());
-
-    assertThat(snapshotChunk.getSnapshotChecksum()).isEqualTo(expectedSnapshotChecksum);
-  }
-
   private ByteBuffer asByteBuffer(final String string) {
     return ByteBuffer.wrap(string.getBytes()).order(Protocol.ENDIANNESS);
   }
 
-  private boolean takeSnapshot(
-      final Path path, final List<String> fileNames, final List<String> fileContents) {
-    assertThat(fileNames).hasSameSizeAs(fileContents);
+  private long getChunkChecksum(final byte[] content) {
+    final var crc32c = new CRC32C();
+    crc32c.update(content);
+    return crc32c.getValue();
+  }
 
+  private boolean takeSnapshot(final Path path) {
     try {
       FileUtil.ensureDirectoryExists(path);
-
-      for (int i = 0; i < fileNames.size(); i++) {
-        final var fileName = fileNames.get(i);
-        final var fileContent = fileContents.get(i);
-        Files.write(
-            path.resolve(fileName), fileContent.getBytes(), CREATE_NEW, StandardOpenOption.WRITE);
-      }
+      SNAPSHOT_CHUNK.forEach(
+          (id, content) -> {
+            try {
+              Files.write(
+                  path.resolve(id), content.getBytes(), CREATE_NEW, StandardOpenOption.WRITE);
+            } catch (final IOException e) {
+              throw new UncheckedIOException(e);
```

Comment:
Nit: you can avoid the nested double IOException try-catch block by avoiding using the lambda but instead just doing a normal `for (final var entry : SNAPSHOT_CHUNK.entrySet())`